modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2516,16 +2516,16 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
         return (reindexed_frames[0], reindexed_frames[1:], joined_index, base_lengths)
 
     @lazy_metadata_decorator(apply_axis="both")
-    def binary_op(self, op, right_frame, join_type="outer"):
+    def n_ary_op(self, op, right_frames: list, join_type="outer"):
```

Comment:
I wonder how this change applies to non-pandas storage formats; also I wonder why I don't see any changes in dask-related execution?

Reply:
> I wonder how this change applies to non-pandas storage formats

Base format doesn't need to fix these bugs according to CI. I don't think we're testing this bug in other storage formats, but I always assume I don't have to worry about that.

The Dask partition manager uses the generic pandas partition manager method I modified here. Ray just wraps that method in a shim so it can apply the progress bar.

Reply:
oh, okay then

Now I only worry if this changes the algebra for the "query compiler spec", and I think we need to either put a fallback `binary_op()` which would call `n_ary_op()` under the hood here, or rename `binary_op()` in other query compilers

cc @modin-project/modin-core @modin-project/modin-omnisci on thoughts

Reply:
I am okay with the renaming of binary op to n_ary op but we should probably change it everywhere.

Reply:
> Now I only worry if this changes the algebra for the "query compiler spec",

@vnlitvinov I don't understand this concern. This PR is changing the API of `PandasDataframe`, and that should only affect the pandas storage format. I thought we would only have to change the pandas query compiler. So we what exactly were you thinking of changing? Would I have to implement `nary_op` in other query compilers?

Reply:
I think we should maintain the same algebra logic (and hence interface level) for all query compilers.

I suppose implementing a wrapper `binary_op()` which would call `n_ary_op()` under the hood would suit me here just fine.

Reply:
We decided offline to keep `n_ary_op` because a lot of the logic in `Binary` and in in functions that use that class in the query compiler is particular to `n=2`.